### PR TITLE
Add virtual root operation

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/tests/test_operations.rs [hunk 1]
```diff
@@ -70,7 +72,7 @@ fn test_op_log() {
     "###);
     insta::assert_snapshot!(
         test_env.jj_cmd_failure(&repo_path, &["log", "--at-op", "@---"]), @r###"
-    Error: The "@---" expression resolved to no operations
+    Error: Workspace "default" doesn't have a working copy
```
Comment (yuja) [inline cli/tests/test_operations.rs:74]: Maybe better to update the query to `@----`. The new error is also interesting, but I think it is the test for empty op

File: lib/src/simple_op_store.rs [hunk 2]
```diff
@@ -69,15 +71,18 @@ impl SimpleOpStore {
     pub fn init(store_path: &Path) -> Self {
         fs::create_dir(store_path.join("views")).unwrap();
         fs::create_dir(store_path.join("operations")).unwrap();
-        SimpleOpStore {
-            path: store_path.to_owned(),
-        }
+        let store = Self::load(store_path);
+        let empty_view_id = store.write_view(&View::default()).unwrap();
+        assert_eq!(store.empty_view_id, empty_view_id);
```
Comment (yuja) [inline lib/src/simple_op_store.rs:76]: Needs a virtual hash?

Suppose we add a new field to the View object, the `empty_view_id` will have to be updated. Then, the existing op store won't have the new empty view object to read.

File: cli/src/operation_templater.rs [hunk 3]
```diff
@@ -148,6 +149,10 @@ fn build_operation_keyword(
             // TODO: introduce dedicated type and provide accessors?
             format!("{}@{}", metadata.username, metadata.hostname)
         })),
+        "root" => {
+            let root_op_id = language.root_op_id.clone();
+            language.wrap_boolean(wrap_fn(move |op| op.id() == &root_op_id))
+        }
```
Comment (yuja) [inline cli/src/operation_templater.rs:155]: nit: docs/templates.md can be updated too.

**Review by yuja — APPROVED** [review]
Nice.

File: cli/tests/test_operations.rs (hunk 1 above)
Comment (martinvonz) [reply cli/tests/test_operations.rs]: Oops, yes (over-aggressive use of `insta test --accept`)
